File: src/main/java/no/nav/fo/veilarbregistrering/enhet/Postadresse.java
package no.nav.fo.veilarbregistrering.enhet;

import no.nav.fo.veilarbregistrering.bruker.Periode;

public class Postadresse implements Adresse {

    private final Kommunenummer kommunenummer;
    private final Periode periode;

    public Postadresse(Kommunenummer kommunenummer, Periode periode) {
        this.kommunenummer = kommunenummer;
        this.periode = periode;
    }

    @Override
    public Kommunenummer getKommunenummer() {
        return kommunenummer;
    }

    @Override
    public boolean erGyldig() {
        return periode.erApen();
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/enhet/Postadresse.kt
package no.nav.fo.veilarbregistrering.enhet;

import no.nav.fo.veilarbregistrering.bruker.Periode;

public class Postadresse implements Adresse {

    private final Kommune kommune;
    private final Periode periode;

    public Postadresse(Kommune kommune, Periode periode) {
        this.kommune = kommune;
        this.periode = periode;
    }

    @Override
    public Kommune getKommunenummer() {
        return kommune;
    }

    @Override
    public boolean erGyldig() {
        return periode.erApen();
    }
}
